libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxCameraModeButton.kt

```diff
@@ -0,0 +1,185 @@
+package com.mapbox.navigation.ui.maps.camera.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import androidx.annotation.DrawableRes
+import androidx.annotation.UiThread
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.mapbox.navigation.ui.maps.R
+import com.mapbox.navigation.ui.maps.camera.state.NavigationCameraState
+import com.mapbox.navigation.ui.maps.databinding.MapboxCameraModeButtonLayoutBinding
+import com.mapbox.navigation.ui.utils.internal.ExtendableButtonHelper
+import com.mapbox.navigation.ui.utils.internal.extensions.measureTextWidth
+
+/**
+ * Default button that allows user to toggle between Camera Following and Overview mode.
+ */
+class MapboxCameraModeButton : FrameLayout {
+
+    private val binding =
+        MapboxCameraModeButtonLayoutBinding.inflate(LayoutInflater.from(context), this)
+
+    private val helper = ExtendableButtonHelper(
+        binding.buttonText,
+        { 0 },
+        { text ->
+            binding.buttonText.measureTextWidth(text).toInt() +
+                resources.getDimensionPixelSize(R.dimen.mapbox_cameraModeButton_paddingStart)
+        },
+    )
+
+    /**
+     * Container view that hosts [iconImage] and [textView].
+     */
+    val containerView: ConstraintLayout = binding.container
+
+    /**
+     * Icon Image.
+     */
+    val iconImage: AppCompatImageView = binding.iconImage
+
+    /**
+     * TextView used to display text when expanded.
+     */
+    val textView: AppCompatTextView = binding.buttonText
+
+    /**
+     * Icon Resource Id for FOLLOWING camera state.
+     */
+    @DrawableRes
+    var followingIconResId: Int = 0
+
+    /**
+     * Icon  Drawable Resource Id for OVERVIEW camera state.
+     */
+    @DrawableRes
+    var overviewIconResId: Int = 0
+
+    /**
+     * Extended mode Text for FOLLOWING camera state.
+     */
+    var followingText: String? = null
+
+    /**
+     * Extended mode Text for OVERVIEW camera state.
+     */
+    var overviewText: String? = null
+
+    /**
+     * Camera state that this button represents.
+     */
+    var cameraState: NavigationCameraState = NavigationCameraState.FOLLOWING
```

Comment:
I don't think we should maintain any state in this `View`. Looking at other UI components, we have veered away from storing states as much as possible for 2 reasons:
- None of our views are capable of retaining states upon config change.
- This adds more logic to the view thereby preventing us from making the view as dumb as possible.

However, there are exceptions where more complicated views do maintain a state in one way or another. I agree that it's tempting to do this to provide ease of integration to the developers, but now that we have DropInUI and it exactly provides a mechanism to do this state management, we might want to keep it there. Past MVP the idea is to move individual `ViewBinders` and `UIComponents` to their respective module and that's where we can do this.

Reply:
This is entirely up to you. This View has been modeled to act more like a [CheckBox](https://developer.android.com/reference/android/widget/CheckBox) then a simple Button.

I will remove `cameraState` and update both `setState` methods to be no longer idempotent.